src/monitor.rs

```diff
@@ -98,48 +22,57 @@ impl std::fmt::Display for VideoModeHandle {
 ///
 /// **Web:** A [`MonitorHandle`] created without
 #[cfg_attr(
-    any(web_platform, docsrs),
+    web_platform,
     doc = "[detailed monitor permissions][crate::platform::web::ActiveEventLoopExtWeb::request_detailed_monitor_permission]."
 )]
-#[cfg_attr(not(any(web_platform, docsrs)), doc = "detailed monitor permissions.")]
+#[cfg_attr(not(web_platform), doc = "detailed monitor permissions.")]
 /// will always represent the current monitor the browser window is in instead of a specific
 /// monitor. See
 #[cfg_attr(
-    any(web_platform, docsrs),
+    web_platform,
     doc = "[`MonitorHandleExtWeb::is_detailed()`][crate::platform::web::MonitorHandleExtWeb::is_detailed]"
 )]
-#[cfg_attr(not(any(web_platform, docsrs)), doc = "`MonitorHandleExtWeb::is_detailed()`")]
+#[cfg_attr(not(web_platform), doc = "`MonitorHandleExtWeb::is_detailed()`")]
 /// to check.
 ///
 /// [`Window`]: crate::window::Window
-#[derive(Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct MonitorHandle {
-    pub(crate) inner: platform_impl::MonitorHandle,
+#[derive(Debug, Clone)]
+pub struct MonitorHandle(pub(crate) Arc<dyn MonitorHandleProvider>);
```

Comment:
Hmm, what if we instead exposed `Arc<dyn MonitorHandle>` to the user, and had `MonitorHandle: PartialEq + Eq + Hash`?

Reply:
Ah, right, because `PartialEq` and `Hash` can't be `dyn`.

Reply:
Maybe there are ways around that?

Reply:
You can, though that only works if monitor equality is implementable on a `u64`, which I don't think it is on macOS?

Reply:
>You can, though that only works if monitor equality is implementable on a u64, which I don't think it is on macOS?

You tell me :- ). In general, an ID is something assigned to you by the display server for that exact reason to compare them. You could have a more _verbose_ monitor IDs.

In general, Though, I don't mind `PartialEq` that way given that you say that it works. Though, can send it _separately_, since you'll be touching other stuff along the lines.